### ISSUE TITLE
errors reported inside `pyproject.toml`/`ruff.toml` are not reported in the language server

### ISSUE THREAD

**Issue by DetachHead** [astral-sh/ruff#27088]
### Summary

```toml
# ruff.toml

target-version = "py314"
preview = true

[lint]
extend-select = ["rule-codes-in-selectors"]
ignore = [
  "N818",
]
```

```console
> ruff check .\ruff.toml
rule-codes-in-selectors: [*] Rule code used instead of name in `lint.ignore`
 --> ruff.toml:7:4
  |
5 | extend-select = ["rule-codes-in-selectors"]
6 | ignore = [
7 |   "N818",
  |    ^^^^
8 | ]
  |
help: Replace rule code with `error-suffix-on-exception-name`
  |
6 | ignore = [
  -   "N818",
7 +   "error-suffix-on-exception-name",
8 | ]
  |

Found 1 error.
[*] 1 fixable with the `--fix` option.
```

in vscode however the error is not reported:

<img width="1004" height="384" alt="Image" src="https://github.com/user-attachments/assets/230421a6-7bb4-4e30-882d-d814d9d00773" />

(that JSON schema error is a separate issue: https://github.com/SchemaStore/schemastore/issues/6082)

output from Ruff Language Server Trace:

```
[Trace - 4:48:00 PM] Sending notification 'workspace/didChangeWatchedFiles'.
Params: {
    "changes": [
        {
            "uri": "file:///c%3A/Users/user/project/ruff.toml",
            "type": 2
        }
    ]
}


[Trace - 4:48:00 PM] Received request 'workspace/diagnostic/refresh - (1)'.
[Trace - 4:48:00 PM] Sending response 'workspace/diagnostic/refresh - (1)'. Processing request took 1ms
No result returned.
```

### Version

ruff 0.15.22 (0177a7e0d 2026-07-16)
